PyInstaller/loader/pyi_archive.py

```diff
@@ -121,7 +158,7 @@ def loadtoc(self):
         self.lib.seek(self.start + self.TOCPOS)
         (offset,) = struct.unpack('!i', self.lib.read(4))
         self.lib.seek(self.start + offset)
-        self.toc = marshal.load(self.lib)
+        self.toc = marshal.loads(self.lib.read())
```

Comment:
Because marshal.load and marshal.dump only support ```file``` object, if pass other object an exception will raised.